src/algorithms/classic/graph_based/potential_field.py

```diff
@@ -97,30 +94,22 @@ def set_display_info(self) -> List[MapDisplay]:
 
 
     def calc_potential_field(self,grid: Map):
-        xw = grid.size.width
-        yw = grid.size.height
 
         # calc each potential
-        pmap = [[0.0 for i in range(yw)] for i in range(xw)]
-
-        for ix in range(xw):
-            #x = ix * self.grid_size + minx
-            x = ix
-
-            for iy in range(yw):
-                #y = iy * self.grid_size + miny
-                y = iy
-                ug = self.calc_attractive_potential(x, y, (grid.goal.position.x), (grid.goal.position.y))
-                if Point(x,y) not in grid.obstacles:
-                    uo = self.calc_repulsive_potential(x=x, y=y, ox=([x.position.x for x in grid.obstacles]), 
-                    oy=([y.position.y for y in grid.obstacles]), rr=(grid.agent.radius))
-                else:
-                    uo = 0
+        pmap = np.zeros(grid.size, dtype=np.float32)
+        #print(str(grid.size))
+
+        for index in np.ndindex(*grid.size):
+            ug = self.calc_attractive_potential(index, grid.goal.position)
+            if Point(*index) not in grid.obstacles:
```

Comment:
is `grid.obstacles` always populated? Maybe best to use `is_agent_valid_pos`, no?